pr_work_done.rb

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env ruby
+# This script for every week will calculate:
+# the number of closed prs
+# the number of merged prs
+# the number of comments made on prs
+
+require 'optparse'
+require 'csv'
+require_relative 'octokit_utils'
+
+options = {}
+options[:oauth] = ENV['GITHUB_COMMUNITY_TOKEN'] if ENV['GITHUB_COMMUNITY_TOKEN']
+parser = OptionParser.new do |opts|
+  opts.banner = 'Usage: pr_work_done.rb [options]'
+
+  opts.on('-n', '--namespace NAME', 'GitHub namespace. Required.') { |v| options[:namespace] = v }
+  opts.on('-r', '--repo-regex REGEX', 'Repository regex') { |v| options[:repo_regex] = v }
+  opts.on('-t', '--oauth-token TOKEN', 'OAuth token. Required.') { |v| options[:oauth] = v }
+
+  # default filters
+  opts.on('--puppetlabs', 'Select Puppet Labs\' modules') {
+    options[:namespace] = 'puppetlabs'
+    options[:repo_regex] = '^puppetlabs-'
+  }
+
+  opts.on('--puppetlabs-supported', 'Select only Puppet Labs\' supported modules') {
+    options[:namespace] = 'puppetlabs'
+    options[:repo_regex] = OctokitUtils::SUPPORTED_MODULES_REGEX
+  }
+
+  opts.on('--community', 'Select community modules') {
+    options[:namespace] = 'puppet-community'
+    options[:repo_regex] = '^puppet-'
+  }
+end
+
+parser.parse!
+
+missing = []
+missing << '-n' if options[:namespace].nil?
+missing << '-t' if options[:oauth].nil?
+if not missing.empty?
+  puts "Missing options: #{missing.join(', ')}"
+  puts parser
+  exit
+end
+
+options[:repo_regex] = '.*' if options[:repo_regex].nil?
+
+util = OctokitUtils.new(options[:oauth])
+repos = util.list_repos(options[:namespace], options)
+
+number_of_weeks_to_show = 10
+
+def date_of_next(day)
+  date  = Date.parse(day)
+  delta = date > Date.today ? 0 : 7
+  date + delta
+end
+
+#find next wedenesday, set boundries
+right_boundry = date_of_next "Wednesday"
+left_boundry = right_boundry - 7
+since = (right_boundry - (number_of_weeks_to_show * 7))
+
+all_merged_pulls = []
+all_closed_pulls = []
+comments = []
+members_of_organisation = {}
+
+#gather all commments / merges / closed in our time range (since)
+repos.each do |repo|
+  closed_pr_information_cache = util.fetch_async("#{options[:namespace]}/#{repo}", git_options={:state=>'closed'}, filter=[:issue_comments], limit={:attribute=>'closed_at', :date=>since})
+  #closed prs
+  all_closed_pulls.concat(util.fetch_unmerged_pull_requests(closed_pr_information_cache))
+  #merged prs
+  all_merged_pulls.concat(util.fetch_merged_pull_requests(closed_pr_information_cache))
+  #find organisation members, if we havent already
+  if members_of_organisation.size == 0
+    members_of_organisation = util.puppet_organisation_members(all_merged_pulls)
```

Comment:
ah, oh, didn't see _that_ :-(


On 17 February 2016 at 14:18, TP Honey <notifications@github.com> wrote:

> In pr_work_done.rb
> <https://github.com/underscorgan/community_management/pull/32#discussion_r53168909>
> :
>
> > +
> > +all_merged_pulls = []
> > +all_closed_pulls = []
> > +comments = []
> > +members_of_organisation = {}
> > +
> > +#gather all commments / merges / closed in our time range (since)
> > +repos.each do |repo|
> > +  closed_pr_information_cache = util.fetch_async("#{options[:namespace]}/#{repo}", git_options={:state=>'closed'}, filter=[:issue_comments], limit={:attribute=>'closed_at', :date=>since})
> > +  #closed prs
> > +  all_closed_pulls.concat(util.fetch_unmerged_pull_requests(closed_pr_information_cache))
> > +  #merged prs
> > +  all_merged_pulls.concat(util.fetch_merged_pull_requests(closed_pr_information_cache))
> > +  #find organisation members, if we havent already
> > +  if members_of_organisation.size == 0
> > +    members_of_organisation = util.puppet_organisation_members(all_merged_pulls)
>
> the code to retrieve members is messy, and requires a pull to get the
> list. really you should just provide the repo. It would need a re-work.
>
> i didnt want to do it in this pr
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/underscorgan/community_management/pull/32/files#r53168909>
> .
>